src/NServiceBus.Core/CriticalError/ConfigureCriticalErrorAction.cs

```diff
@@ -17,7 +17,7 @@ public static void DefineCriticalErrorAction(this EndpointConfiguration endpoint
         {
             Guard.AgainstNull(nameof(endpointConfiguration), endpointConfiguration);
             Guard.AgainstNull(nameof(onCriticalError), onCriticalError);
-            endpointConfiguration.Settings.Set("onCriticalErrorAction", onCriticalError);
+            endpointConfiguration.Settings.Get<HostingComponent.Configuration>().CustomCriticalErrorAction = onCriticalError;
```

Comment:
looks like SC might be using the settings key directly, but that shouldn't be an issue from what I can see, right? https://github.com/search?q=org%3AParticular+%22onCriticalErrorAction%22&type=Code

Reply:
Yes, we should pick that up via https://github.com/Particular/NServiceBus/pull/5492/files#diff-a2452b412bdc7681e72991746302fcc1R140

I've added a note to smoke test SC in our POA